capsules/src/lps25hb.rs

```diff
@@ -171,27 +196,88 @@ impl i2c::I2CClient for LPS25HB<'_> {
             }
             State::TakeMeasurementInit => {
                 buffer[0] = Registers::PressOutXl as u8 | REGISTER_AUTO_INCREMENT;
-                // TODO verify errors
-                let _ = self.i2c.write(buffer, 1);
-                self.state.set(State::TakeMeasurementClear);
+                if let Err((error, buffer)) = self.i2c.write(buffer, 1) {
+                    self.state.set(State::Idle);
+                    self.buffer.replace(buffer);
+                    self.owning_process.map(|pid| {
+                        let _ = self.apps.enter(*pid, |_app, upcalls| {
+                            upcalls
+                                .schedule_upcall(
+                                    kernel::into_statuscode(Err(error.into())),
```

Comment:
The first argument is the index of the upcall. Since there is only one upcall, this should always be 0.